### DISCUSSION TITLE
Remove myself as reviewer from pull requests

### DISCUSSION THREAD

**Discussion by loic-sharma** [community/community#16730] [Pull Requests]
Hello! 👋 

## Problem

I would like to use the "Review Requests" page to track what needs my attention: https://github.com/pulls/review-requested

I've changed companies and lost permissions to repositories that are owned by my previous employer. However, I'm still listed as a reviewer on several stale pull requests. I do not have permissions to close these stale pull requests, or, to remove myself as a reviewer. As a result, the "Review Requests" page is polluted with items I cannot remove myself.

## Suggestion

Allow users to remove themselves as a requested reviewer from a pull request even if they do not have permissions for that repository.

See this discussion: https://github.community/t/how-to-remove-a-reviewer-from-a-pull-request/1415

**Comment by ghost** [comment]
Hey! I found a following way:
- open a PR
- locate the list of PR reviewers
- on top of the list there's "Reviewers" button with a "gear" icon. Click on it
- in the drop-down list of the reviewers locate yourself
- uncheck yourself from the list, this should remove you from the list
- repeat for the other PRs

**Reply by loic-sharma** [reply]
This works only if you have permissions to add/remove reviewers from a pull request. However, it doesn't work if you lost permissions to a repository, which can happen if you change companies.

**Reply by tom-nisbet** [reply]
Still appears broken in my testing. I was able to remove myself, but I can't remove others that I added. The reviewers I added also can't remove themselves, effectively orphaning them as reviewers on a PR that they won't aren't required for.

**Reply by joshterrill** [reply]
Same issue here. I am an admin for an organization, and I accidentally added the wrong person to a PR as a reviewer, and now I have no way of removing them. 

I found this article here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review

But it appears to be out of date as I don't see any options like that when reviewing a PR reviewer. And in the settings cog as is discussed above, there is no option to remove a reviewer that has already been added that is currently in a pending state, if it isn't you.

**Reply by jsoref** [reply]
I see that feature:
![image](https://user-images.githubusercontent.com/2119212/182961779-2ec878f2-3c04-47cd-b186-8c2ed7f3845e.png)

And I confirmed that a member w/ write is able to use the feature too...

Note that I had to click `Show all reviewers` / expand the approval to show an item w/ the ⋯.

**Reply by loic-sharma** [reply]
@jsoref do you have contributor access to the repository? I don't have the `...` drop down on repositories that I've lost permissions, here's what it looks like:

![image](https://user-images.githubusercontent.com/737941/183128771-8a481e62-c9ac-4354-bbdb-b1d50ebc1b15.png)

**Reply by jsoref** [reply]
I was replying to @joshterrill.

Offhand, it _looks_ like you could technically "request changes" or "approve". Obviously neither of these are great, but, a change of "please get me out of here, I'm trapped" is a kinda valid response.

Fwiw, I gave up and filed a ticket based on this issue. I agree that one should be able to quietly leave the request list w/o having Triage or better permissions.

**Reply by joshterrill** [reply]
@jsoref that's correct, the whole process is weird. I've actually taken a video recording showing what I mean. Here is me, being an admin of an organization, and an admin of a repo, starting to create a PR, adding someone, then I can't remove them, I can only add another person. Then I create the PR, and I still can't remove them, I can only change who it is assigned to. When all I want to do is just remove the reviewer all together. See attached video.

https://user-images.githubusercontent.com/5368757/183142095-34b72494-4085-4447-af55-48764636ec76.mp4

**Reply by jsoref** [reply]
So. I suspect that it you pushed an extra commit to the branch, you'd be able to use the dismiss stale requests bit. 

I should have taken a video of mine. I'll have to get permission from someone.

At the end, after you've created the PR, can you still not remove the current requestee from the right sidebar?

Also, is it relevant that your PR has a merge conflict?

(I used a brand new repository with only two commits for testing.)

**Reply by joshterrill** [reply]
@jsoref 
1. I just created a test commit in a branch to test this, so the commit was made just before the PR
2. Yes, after I create the PR, I still cannot remove the requestee from the sidebar, this is shown in my video at ~0:49
3. In my video, it had a conflict, when I discovered this and what led me to comment, there was no conflict. If you want me to reproduce this and show you without a conflict, I can, but I get the same results.

**Reply by jsoref** [reply]
Hmm, this is fun. -- If a reviewer no longer has `read` permission, the 🔄 widget doesn't do anything useful and doesn't explain that it doesn't work.

Here's me trying to show that it's possible occasionally to do things:
https://user-images.githubusercontent.com/2119212/183197671-437f39d3-5f63-44f6-b942-51fdb2051cb2.mov

I think I'd need to reduce my screen res (this was actually a tiny window when I recorded it, but...).

Note that in my case the reviewer in question actually had created a review -- we played w/ it where they hadn't as well. The UI is clearly quite rough and hasn't had enough complaints to force it into a particularly smooth state.

**Comment by parberge** [comment]
https://github.com/orgs/community/discussions/23054 also speaks about the problem that a user show up as a reviewer when you write a comment, and that user can't be removed from the list even if it's not a required reviewer.

Will this feature request cover that use case also?

**Reply by holmbergjonas** [reply]
Yes, it will also be added to things never to do or speak of :+1:

**Comment by nusrantham** [comment]
remove  Cloning a repository
